src/public/server.js

```diff
@@ -223,195 +236,121 @@ Dungeon.prototype = {
     });
 
   },
-  applyOption: function (x, y, optionName, bully) {
-    switch (optionName) {
-      case 'wall':
-        {
-          if (this.area[x][y].state === 'square' &&
-            bully.money >= this.config.wallCost &&
-            this.id !== bully.id) {
-            this.area[x][y].state += ' wall';
-          }
-          break;
-        }
-      case 'trap':
-        {
-          if (this.area[x][y].state === 'square' &&
-            bully.money >= this.config.trapCost &&
-            this.id !== bully.id) {
-            this.player.trapped = true;
-            this.area[x][y].state += ' trap';
-          }
-          break;
-        }
+  // return true if state applyed, false otherwise
+  applyState: function (x, y, requestedState, bullyDungeon) {
+
+    if(this.id === bullyDungeon.id) return false;
+
+    var originalState = this.area.getState(x, y);
+    
+    if( (requestedState & STATE_WALL) && (originalState & STATE_DEFAULT) && (bullyDungeon.money >= this.config.wallCost) ) {
+      this.area.setState(x, y, originalState | requestedState);
+      return true;
+    } else if ( (requestedState & STATE_TRAP) && (originalState & STATE_DEFAULT) && (bullyDungeon.money >= this.config.trapCost) ) {
+      this.area.setState(x, y, originalState | requestedState);
+      return true;
     }
+    
+    return false;
   },
-  deduceMoney: function (x, y, optionName, victim) {
-    var moneyToDeduce = 0;
-    switch (optionName) {
-      case 'wall':
-        {
-          if (victim.area[x][y].state === 'square') {
-            if (this.money >= this.config.wallCost) moneyToDeduce += this.config.wallCost;
-          }
-          break;
-        }
-      case 'trap':
-        {
-          if (victim.area[x][y].state === 'square') {
-            if (this.money >= this.config.trapCost) moneyToDeduce += this.config.trapCost;
-          }
-          break;
-        }
+  // This method should not be called if player doesn't have enough money
+  deduceMoney: function (requestedState) {
+    /// @TODO make a global cell state object to centralize label, cost, etc.
```

Comment:
this TODO seems done.
As the config for the costs are in the dungeons and the global state variables are done.
maybe remove ?

Reply:
Yes, as we discussed a bit, i am not sure about this idea anymore. I won't do a commit just to remove this comment but i'll consider it in coming developments.